packages/dd-trace/src/appsec/rasp/ssrf.js

```diff
@@ -25,16 +41,108 @@ function analyzeSsrf (ctx) {
 
   if (!req || !outgoingUrl) return
 
+  // Determine if we should collect the response body based on sampling rate
+  const includeBodies = downstream.shouldSampleBody(req)
+
+  // Initialize state for tracking this request's response
+  ctx[RESPONSE_STATE] = {
+    req,
+    includeBodies,
+    chunks: includeBodies ? [] : null,
+    done: false
+  }
+
+  const requestAddresses = downstream.extractRequestData(ctx)
+
   const ephemeral = {
-    [addresses.HTTP_OUTGOING_URL]: outgoingUrl
+    [addresses.HTTP_OUTGOING_URL]: outgoingUrl,
+    ...requestAddresses
   }
 
-  const raspRule = { type: RULE_TYPES.SSRF }
+  const raspRule = { type: RULE_TYPES.SSRF, variant: 'request' }
 
   const result = waf.run({ ephemeral }, req, raspRule)
 
-  const res = store?.res
-  handleResult(result, req, res, ctx.abortController, config, raspRule)
+  handleResult(result, req, store?.res, ctx.abortController, config, raspRule)
+
+  downstream.incrementDownstreamAnalysisCount(req)
+
+  // Track body analysis count if we're sampling the response body
+  if (includeBodies) {
+    downstream.incrementBodyAnalysisCount(req)
+  }
+}
+
+/**
+ * Collects outgoing response chunks when body sampling is enabled.
+ * @param {{
+ * ctx: object,
+ * chunk: Buffer|string|Uint8Array,
+ * res: import('http').IncomingMessage}} payload event payload from the channel.
+ */
+function handleResponseData ({ ctx, chunk, res }) {
+  if (!res || !chunk) return
+
+  const state = ctx[RESPONSE_STATE]
+
+  if (!state?.includeBodies || state?.done) return
+
+  // Handle both string chunks (from setEncoding) and Buffer chunks
+  if (typeof chunk === 'string') {
+    state.chunks.push(chunk)
+  } else if (Buffer.isBuffer(chunk)) {
+    state.chunks.push(chunk)
+  } else {
+    // Handle Uint8Array or other array-like types
+    state.chunks.push(Buffer.from(chunk))
+  }
+}
+
+/**
+ * Finalizes body collection for the response and triggers RASP analysis.
+ * @param {{ctx: object, res: import('http').IncomingMessage}} payload event payload from the channel.
+ */
+function handleResponseFinish ({ ctx, res }) {
+  if (!res) return
+
+  const state = ctx[RESPONSE_STATE]
+  if (!state || state.done) return
+
+  state.done = true
+
+  // Combine collected chunks into a single body (or null if no chunks)
+  let body = null
+  if (state.chunks?.length) {
+    const firstChunk = state.chunks[0]
+    body = typeof firstChunk === 'string'
+      ? state.chunks.join('')
+      : Buffer.concat(state.chunks)
+  }
```

Comment:
if we move the logic in `handleResponseData` to client.js, this logic would be also there, and body would be send directly as parameter in the finish channel.